src/runtime_src/core/common/system.h

```diff
@@ -140,6 +140,12 @@ class system
   {
     throw std::runtime_error("memory write is not supported");
   }
+
+  virtual void
+  mem_write(const device*, long long, long long, std::vector<char>) const
+  {
+    throw std::runtime_error("memory write is not supported");
+  }
```

Comment:
Why is this function added to system.h?   
I see the implementation being nothing but some query requests.  I don't think it belongs here.
Nor do I think any of the earlier mem_write/mem_read belong here.

In terms of just the signature, then the std::vector<char> should definitely not be passed by value.

Reply:
Hopefully the changes I made fit the bill, I removed all references to write/read from the system files and reference these type of functions from memaccess.h instead